### PR TITLE
CTRL+C sometimes throws a JsonEOFException from the CLI #2640

### LINKED ISSUE #2640: CTRL+C sometimes throws a JsonEOFException from the CLI
The CLI sometimes throw an ugly exception if CTRL+C is pressed while a `SELECT` query is displaying stream results.

```
ksql> select * from PAGEVIEWS_ORIGINAL;
1554404838519 | 16051 | 1554404838519 | User_5 | Page_50
1554404838543 | 16061 | 1554404838542 | User_9 | Page_21
1554404838559 | 16071 | 1554404838559 | User_8 | Page_20
1554404838618 | 16081 | 1554404838618 | User_9 | Page_65
1554404838629 | 16091 | 1554404838629 | User_9 | Page_16
1554404838643 | 16101 | 1554404838643 | User_8 | Page_20
^CQuery terminated
com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input in field name
 at [Source: (String)"{"row":{"columns":[1554404838713,"16111",1554404838713,"User_7","Page_90"]},"errorMess"; line: 1, column: 173]
Caused by: Unexpected end-of-input in field name
 at [Source:
	(String)"{"row":{"columns":[1554404838713,"16111",1554404838713,"User_7","Page_9
	0"]},"errorMess"; line: 1, column: 173]
```

To reproduce it, run the the `pageviews` datagen examples (or just execute this script https://github.com/confluentinc/examples/blob/5.2.1-post/pageviews/start.sh)

Then log in to KSQL and execute the `SELECT` query mentioned above. It does not fail every time, so you might need to run it a few times and press `CTRL+C`.

### PR DESCRIPTION
**Pull request by ouertani**
### Description 
This fixes #2640

### Testing done 
I add some exception.printStackTrace()
https://github.com/confluentinc/ksql/blob/74e813f384088c097cf623a99d032861bff2ccb1/ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java#L380
That sometimes reproduce the exception.
### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by apurvam** [comment]
Thanks for the patch @ouertani . Did you do do any tests to validate that your change fixes the problem? If so, mentioning that in the PR description would be good.

**Comment by ouertani** [comment]
@apurvam This is a random behaviour due to concurrency between the close and stream the data. I add some exception.printStackTrace() 
[here](https://github.com/confluentinc/ksql/blob/74e813f384088c097cf623a99d032861bff2ccb1/ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java#L380)
That sometimes reproduce the exception.

**Comment by spena** [comment]
Thanks for taking this issue @ouertani I left a comment on the code. I think we need to have more `synchronized` blocks in the class to be thread-safe.

Btw, it would be good to add more information on the description of this issue. It's a common pattern we have when submitting a PR. If you could provide a little description of the issue and how it is fixed. Also, what tests you made.

It is good to mention the Issue fixed as well at the beginning of the PR description (before the Description section), like this:
This fixes https://github.com/confluentinc/ksql/issues/2640

The above will make sure to close the issue automatically when this PR is merged. 

One more thing, could you make the changes for `5.2.x` instead of master?  Once the code is merged, I can merge the changes manually to master.

File: docs/installation/server-config/config-reference.rst [hunk 1]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
 the prefix for the internal topics created by KSQL. Using the default value ``ksql.service.id``, the KSQL internal topics
 will be prefixed as ``_confluent-ksql-default_`` (e.g. ``_command_topic`` becomes ``_confluent-ksql-default__command_topic``).
 
+.. _ksql-internal-topic-replicas:
+
+--------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
----------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 2]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
+ksql.internal.topic.replicas
+--------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
----------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 3]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
+
+The number of replicas for the internal topics created by KSQL Server. The default is one.
+This config parameter works in KSQL 5.3 and later.
+Note that replicas for record processing log topic should be configrued separately. For more info refer to :ref:`KSQL Processing Log <ksql_processing_log>`.
+
 .. _ksql-sink-partitions:
 
 --------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
---------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 4]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
-ksql.sink.partitions
+ksql.sink.partitions (Deprecated)
 --------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
---------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 5]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
 
 The default number of partitions for the topics created by KSQL. The default is four.
+This property has been deprecated since 5.3 release. For more info see the WITH clause properties in :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` and :ref:`CREATE TABLE AS SELECT <create-table-as-select>`.
 
 .. _ksql-sink-replicas:
 
 ------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
-------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 6]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
-ksql.sink.replicas
+ksql.sink.replicas (Deprecated)
 ------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
-------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 7]
```diff
@@ -225,10 +237,10 @@ listeners
 ---------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
------------------------------------
```

**Comment by spena** [comment]
@ouertani There are 1K files modified. I think you just changed the target to be 5.2.x, That happened to me once. What you will need to do is to create your branch based from 5.2.x, and then apply your changes there.

**Comment by ouertani** [comment]
@spena Supersedes  https://github.com/confluentinc/ksql/pull/2730

**Pull request closed without merging**